### PR TITLE
feat: add settings for OpenTelemetry export

### PR DESCRIPTION
**Pull request by TBonnin**
## Describe your changes

Add environment settings to input OpenTelemetry endpoint and headers
Screenshot 2024-10-30 at 13 22 26

Note: 
- an empty endpoint is considered as disabled. I don't think adding an extra checkmark is actually useful
- The OpenTelemetry feature is only enabled for enterprise setup or via feature flag. I was thinking I would hide the settings if not enabled but I think it is nice to promote the feature even if not enabled. Editing the settings would show `OpenTelemetry export is not enabled for this account` error. I am happy to also make a mention of it being a paid feature in the environment setting page. Let me know @bastienbeurier 

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/webapp/src/pages/Environment/Settings.tsx
Comment (TBonnin) [inline packages/webapp/src/pages/Environment/Settings.tsx:129]: caveat: headers are reordered by key automatically. I didn't want to make the model and code more complicated for such a tiny feature and the fact that in most case there will be only 1 header max

File: packages/server/lib/controllers/v1/environment/otlp/postSettings.ts
Comment (bodinsamuel) [inline packages/server/lib/controllers/v1/environment/otlp/postSettings.ts:1]: having one endpoint per setting is a bit insane imo
what do you think of taking this opportunity to create a generic endpoint (that only accepts otlp for now)?

File: packages/shared/lib/services/environment.service.ts
Comment (bodinsamuel) [inline packages/shared/lib/services/environment.service.ts:341]: same for this, there are 20 update functions while only need one that does this
```
await db.knex.from<DBEnvironment>(TABLE).where({ id: environmentId }).update(updatedEnv);
```

File: packages/webapp/src/pages/Environment/Settings.tsx
Comment (bodinsamuel) [inline packages/webapp/src/pages/Environment/Settings.tsx:1166]: Would be nice to have an example in the tooltip or better in the placeholder of the input

File: packages/server/lib/controllers/v1/environment/otlp/postSettings.ts
Comment (bodinsamuel) [inline packages/server/lib/controllers/v1/environment/otlp/postSettings.ts]: is it a URL?

File: packages/webapp/src/pages/Environment/Settings.tsx
Comment (bodinsamuel) [inline packages/webapp/src/pages/Environment/Settings.tsx:1061]: To see with @bastienbeurier but I feel like it would be better for the UX to clearly state that the account has no access to this feature instead of clicking edit, pasting something, and then getting an error.
You can return the flag in the `/meta`.

File: packages/server/lib/controllers/v1/environment/otlp/postSettings.ts
Comment (bodinsamuel) [inline packages/server/lib/controllers/v1/environment/otlp/postSettings.ts:52]: what about self hosted?

**Review by bodinsamuel — COMMENTED** [review]
few open questions

File: packages/server/lib/controllers/v1/environment/otlp/postSettings.ts
Comment (TBonnin) [reply packages/server/lib/controllers/v1/environment/otlp/postSettings.ts:1]: it is a bit insane. I also think the design is now overwhelming with all those options and I would vote for not only a endpoints refactor but a UI redesign as well. cc @bastienbeurier 
I would prefer to push this otlp export feature to completion though and not go into a refactor at this point.

File: packages/webapp/src/pages/Environment/Settings.tsx
Comment (TBonnin) [reply packages/webapp/src/pages/Environment/Settings.tsx:1166]: good point. will do

File: packages/server/lib/controllers/v1/environment/otlp/postSettings.ts
Comment (TBonnin) [reply packages/server/lib/controllers/v1/environment/otlp/postSettings.ts]: yes. good point

File: packages/server/lib/controllers/v1/environment/otlp/postSettings.ts
Comment (TBonnin) [reply packages/server/lib/controllers/v1/environment/otlp/postSettings.ts:52]: it is supposed to be a scale plan only feature so they can only be enterprise or cloud. Am I missing something?

File: packages/webapp/src/pages/Environment/Settings.tsx
Comment (TBonnin) [reply packages/webapp/src/pages/Environment/Settings.tsx:1061]: will check with @bastienbeurier

File: packages/server/lib/controllers/v1/environment/otlp/postSettings.ts
Comment (bodinsamuel) [reply packages/server/lib/controllers/v1/environment/otlp/postSettings.ts:52]: it shouldn't be shown in the UI then

File: packages/webapp/src/pages/Environment/Settings.tsx
Comment (TBonnin) [reply packages/webapp/src/pages/Environment/Settings.tsx:1061]: as per our discussion with @bastienbeurier: yes this would be a better UX but let's ship it like this for now and we will work on redesigning the env settings page including clearly stating that this option is paid only

File: packages/server/lib/controllers/v1/environment/otlp/postSettings.ts
Comment (TBonnin) [reply packages/server/lib/controllers/v1/environment/otlp/postSettings.ts:52]: Agreed. Just check with @bastienbeurier though and we decided to keep it as is for now (aka: failing unless enterprise or cloud with flag enabled) and improve it when making the design of this page better

**Comment by bodinsamuel** [comment]
let me know when you need a second review

**Comment by TBonnin** [comment]
@bodinsamuel fixed the zod schema and the endpoint placeholder.
The rest (multiple edit endpoints and functions, ux, etc...) will be improved/refactored later.

**Review by bodinsamuel — APPROVED** [review]

**Review by bodinsamuel — APPROVED** [review]

**Pull request merged**
